src/internal_modules/roc_packet/rtp.h

```diff
@@ -45,6 +46,15 @@ struct RTP {
     //!  Duration is measured in the same units as timestamp.
     timestamp_t duration;
 
+    //! Timestamp of the first sample at the moment it was captured from an interface.
+    //! @remarks
+    //!  In an ideal case the meaning of this value should be the same on a sender
+    //!  and a receiver, particularly it should store the moment in time the first sample
+    //!  of a packet came into existence. This could be achieved by outbound signaling
+    //!  such as RTCP (with XR extension). In case RTCP is not fully functional
+    //!  this value could store the moment a packet arrived.
+    core::nanoseconds_t capture_timestamp;
```

Comment:
I think capture_timestamp is not really RTP-specific, unlike other fields in this struct. I can imagine that we will somehow fill it for other protocols too. Also its format and usage are protocol-independent, and it does not correspond to anything in RTP header. So I suggest to move it to Packet class.

Reply:
Every time we interact with capture timestamp in the code we rely on number of samples in the packet. So far `capture_timestamp` is strictly bounded to RTP (with an audio stream).